caffe2/core/nomnigraph/Converters/Caffe2.cc

```diff
@@ -0,0 +1,327 @@
+#include "nomnigraph/Converters/Caffe2.h"
+#include "nomnigraph/Graph/Algorithms.h"
+
+#include "nomnigraph/Support/Casting.h"
+#include "nomnigraph/Support/Pointer.h"
+
+std::map<std::string, caffe2::Argument>
+getArgumentsFromOperator(caffe2::OperatorDef op) {
+  std::map<std::string, caffe2::Argument> argMap;
+  for (auto arg : op.arg()) {
+    argMap[arg.name()] = arg;
+  }
+  return argMap;
+}
+
+namespace nom {
+namespace converters {
+
+std::unique_ptr<repr::NeuralNetOperator>
+convertOperator(caffe2::OperatorDef op) {
+  auto argMap = getArgumentsFromOperator(op);
+
+  if (op.type() == "Conv") {
+    // There are literally three ways to define shapes in Conv in Caffe2
+    std::vector<int> kernelShape;
+    if (argMap.count("kernel")) {
+      int kernel = static_cast<int>(argMap["kernel"].i());
```

Comment:
It is worth to add e.g. `CAFFE_ENFORCE(argMap["kernel"].has_i(), "Error message");` before all lines that access `ArgumentProto` fields. Technically, these fields are optional in the message.